pylearn2/models/mlp.py

```diff
@@ -3059,6 +3079,14 @@ def set_input_space(self, space):
 
         rng = self.mlp.rng
 
+        if self.pad != (0, 0):
+            output_shape = \
+                [int(np.ceil((i_sh + 2. * k_pad - k_sh) / float(k_st))) + 1
```

Comment:
(I got rid of pad, following Fred's suggestion, so this comment does not apply anymore to this PR. I keep it for future reference.)

If pad is (0, 0) this is not executed.